ibis/expr/types/generic.py

```diff
@@ -475,52 +475,49 @@ def __le__(self, other: Value) -> ir.BooleanValue:
     def __lt__(self, other: Value) -> ir.BooleanValue:
         return _binop(ops.Less, self, other)
 
-
-@public
-class Scalar(Value):
     def to_projection(self) -> ir.Table:
-        """Promote this scalar expression to a projection."""
+        """Promote this value expression to a projection."""
         from ibis.expr.analysis import find_immediate_parent_tables
-        from ibis.expr.types.relations import Table
 
         roots = find_immediate_parent_tables(self.op())
         if len(roots) > 1:
             raise com.RelationError(
-                'Cannot convert scalar expression '
+                f'Cannot convert {type(self)} expression '
                 'involving multiple base table references '
                 'to a projection'
             )
 
-        table = Table(roots[0])
-        return table.projection([self])
+        return roots[0].to_expr().projection([self])
+
+
+@public
+class Scalar(Value):
+    def __rich_console__(self, console, options):
+        return console.render(str(self.execute()), options=options)
 
     def _repr_html_(self) -> str | None:
         return None
 
 
 @public
 class Column(Value):
-    def to_projection(self) -> ir.Table:
```

Comment:
Did we leave it from previous refactors?

Reply:
Maybe? I can't remember.